### ISSUE TITLE
UIScrollToSelection vertical snap problem / fix?

### ISSUE THREAD

**Issue by SimonDarksideJ** [Unity-UI-Extensions/com.unity.uiextensions#184]
<sup>Issue created by Jonathan Menard as [Bitbucket Issue #&#8203;148](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/148) on 2017.06.21 04:18.</sup>
I was having issues scrolling up in a large list of items with UIScrollToSelection so I had to change the code to this:

```C#

        private float GetScrollOffset(float position, float listAnchorPosition, float targetLength, float maskLength) {
            if (position < listAnchorPosition + (targetLength/2)) {
                return (listAnchorPosition + maskLength) - (position - targetLength);
            } else if (position + targetLength > listAnchorPosition + maskLength) {
                return (listAnchorPosition + maskLength) - (position + targetLength);
            }

            return 0;
        }
```

This fixes the issue for now, but it "snaps" up instead of the smooth scroll. Not sure what is causing the issue. Not sure why this works for me / maybe not others. My item pivots are x=0.5 y=0.5. Maybe that needs to be factored in?

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.06.22 22:21](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/148#comment-37783480), @SimonDarksideJ commented:</sup>
Thanks for the help. will look in to adding this for the upcoming update.

**Comment by SimonDarksideJ** [comment]
<sup>On [2017.07.02 14:44](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/148#comment-37972387), @SimonDarksideJ commented:</sup>
Adopts fix from #148 Scroll To improved Large list support

Resolves #148


Referenced by 5cebd6cdfe27

**Comment by SimonDarksideJ** [comment]
<sup>On 2017.07.02 14:44 @SimonDarksideJ modified issue:</sup>
**status** changed `new` → `resolved`
